blocks/bricks/__init__.py

```diff
@@ -464,6 +464,10 @@ def apply(self, input_):
             child_input = output
         return output
 
+    @apply.property('outputs')
```

Comment:
It seems that all you need is a) make `@apply.property('inputs')` b) accept `*args` in `apply c) remove `inputs` from `@application` declaration.